zeppelin-zengine/src/main/java/org/apache/zeppelin/notebook/repo/ipfs/Ipfs.java

```diff
@@ -0,0 +1,220 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.zeppelin.notebook.repo.ipfs;
+
+import com.google.gson.Gson;
+import com.google.gson.reflect.TypeToken;
+
+import org.apache.http.HttpEntity;
+import org.apache.http.HttpStatus;
+import org.apache.http.client.config.RequestConfig;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.entity.ContentType;
+import org.apache.http.entity.mime.MultipartEntityBuilder;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClients;
+import org.apache.http.util.EntityUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.lang.reflect.Type;
+import java.util.Map;
+
+
+/**
+ * Ipfs HttpClient
+ */
+public class Ipfs {
+  private static final Logger LOG = LoggerFactory.getLogger(Ipfs.class);
+  private CloseableHttpClient client;
+  private String baseUrl;
+
+  public Ipfs(String url) {
+    baseUrl = url;
+    client = HttpClients.createDefault();
+  }
+
+  public String pinLs() {
+    String pinnedJsonString = null;
+    HttpGet request = new HttpGet(baseUrl + "pin/ls?type=recursive&quiet=true");
+    try {
+      CloseableHttpResponse response = client.execute(request);
+      try {
```

Comment:
In the inner try `response.close()` throws a IOException. If I have a single try block,then the finally block will be outside and there will be try catch in finally block for `reponse.close()`.
```
finally {
      try {
        response.close();
      } catch (IOException | NullPointerException e) {
        LOG.error("Failed to close response", e);
      }
```
Is this the right way ?